Library/SPIClient.m

```diff
@@ -1201,7 +1201,6 @@ - (void)handleGetLastTransactionResponse:(SPIMessage *)m {
             } else {
                 // TH-4X - Unexpected Error when recovering
                 SPILog(@"Unexpected response in get last transaction during - received posRefId:%@ error: %@", [gltResponse getPosRefId], m.error);
-                [txState unknownCompleted:@"Unexpected error when recovering transaction status. Check EFTPOS."];
```

Comment:
you need to replace this with a `return` statement so we don't unnecessarily   publish the state change.

Reply:
Added.